### ISSUE TITLE
Accessibility issues: keyboard trap in entry window

### ISSUE THREAD

**Issue by LittleStar-VIP** [keepassxreboot/keepassxc#9060]
## Overview
Hello,
This is a screen reader user, who rely on screen reader to navigate among the application gui elements, and I found that I have problem in interacting with some of the elements

## Steps to Reproduce
[NOTE]: # ( Provide a simple set of steps to reproduce this bug. )
1. create a new entry or edit an existing entry
2. use keyboard tab key to tab to the note field
3. if I press another tap key, my screen reader become silent, and cannot interact with any of the element in the entry window
4. pressing shift + tab also unable to interact with any of the element

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen )
Using tab should be able to cycle through elements on the window without trapping in one particular element

## Actual Behavior
[NOTE]: # ( Tell us what actually happens )
focus is being trapped

## Context
[NOTE]: # ( Give us any additional information you may have. )
The screen reader that I am using is NVDA 2022.4

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - VERSION
KeePassXC 2.7.4

[NOTE]: # ( Pick choices based on your environment )
Operating System: Windows
Desktop Env: not sure what is this
Windowing System: Windows 10 Home ed

**Comment by droidmonkey** [comment]
Unfortunately there is not much we can do about that since we allow tabs on notes. Shift+TAB should bring you back to the previous ui element.

**Comment by LittleStar-VIP** [comment]
Hmm, I have tried tab and shift tab, both don't work.

**Comment by michaelk83** [comment]
> Unfortunately there is not much we can do about that since we allow tabs on notes.

https://stackoverflow.com/a/53910297/15047974 - this is for html textareas, but similar principles apply here. The simple solution is to use Esc (or Ctrl+M) to exit text editing mode, and resume the tabbing sequence. Since Esc is also used for the Cancel button, this would need to be overridden while editing text: 1st Esc exits the text edit, 2nd Esc activates the Cancel button (maybe with some delay, to avoid accidental double-press issues). Another recommendation is to not capture Tab presses until something is typed into the text area.

**Comment by droidmonkey** [comment]
Oooh that inspired me, this actually worked great:
![image](https://user-images.githubusercontent.com/2809491/216331370-eb1fc99e-f4c4-4f1f-9348-e0ea16189e51.png)
